### PR TITLE
Feature/issue 02

### PR DESCRIPTION
**Pull request by kevin-verschaeve**
https://github.com/poledev/kevin/issues/6

### CODE REVIEW

**Comment by mRoca** [comment]
Your branch has conflicts with the target one, you must rebase it.

File: app/config/config.yml [hunk 1]
```diff
@@ -72,3 +72,10 @@ swiftmailer:
     #scope:          %sensiolabs_connect.scope%
     #oauth_endpoint: %sensiolabs_connect.oauth_endpoint%
     #api_endpoint:   %sensiolabs_connect.api_endpoint%
+
+stof_doctrine_extensions:
+    default_locale: en_EN
+    orm:
+        default:
+            timestampable: true
+            sluggable: true
```
Comment (mRoca) [inline app/config/config.yml]: Fix CS (you can use the [PHP-CS-Fixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer) tool)

File: src/SensioLabs/JobBoardBundle/Controller/JobController.php [hunk 2]
```diff
@@ -19,12 +20,26 @@ public function showAction()
...
-     * @Route("/preview", name="job_preview")
+     * @Route("/{countryCode}/{contractType}/{slug}/preview", name="job_preview")
      * @Template()
      */
-    public function previewAction()
+    public function previewAction($countryCode, $contractType, $slug)
     {
-        return array();
+        $jobRepo = $this->getDoctrine()
+                        ->getManager()
```
Comment (mRoca) [inline src/SensioLabs/JobBoardBundle/Controller/JobController.php]: `getManager()` not needed

File: app/AppKernel.php [hunk 3]
```diff
@@ -20,6 +20,7 @@ public function registerBundles()
             // new SensioLabs\Bundle\ConnectBundle\SensioLabsConnectBundle(),
             // new SensioLabs\Internal\Bundle\ConnectBundle\SensioLabsInternalConnectBundle(),
```
Comment (mmarchois) [inline app/AppKernel.php]: Always remove comments

File: src/SensioLabs/JobBoardBundle/Form/Type/JobType.php [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            ->add('city', 'text', array(
+                'required' => true,
+                'attr' => array(
+                    'placeholder' => 'City',
+                    'maxlength' => 80
+                )
+            ))
+            ->add('contractType', 'choice', array(
+                'choices' => array(
+                    'CDI' => 'Full Time',
```
Comment (mRoca) [inline src/SensioLabs/JobBoardBundle/Form/Type/JobType.php]: You could use static functions and/or constants on your entities to store the static values lists.

File: src/SensioLabs/JobBoardBundle/Form/Type/JobType.php [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+                )
+            ))
+        ;
+    }
+
+    public function getName()
+    {
+        return 'jobboard_post';
+    }
+}
```
Comment (mmarchois) [inline src/SensioLabs/JobBoardBundle/Form/Type/JobType.php]: PHP CS

File: src/SensioLabs/JobBoardBundle/Resources/views/Includes/full_job_header.html.twig [hunk 6]
```diff
@@ -1,17 +1,17 @@
...
 <div class="details">
     <span class="filter">
-        <a href="#">Paris, France</a> /
-        <a href="#">Full time</a>
-        
-        <br/><span class="archived">Archived since 12/12/2013</span>
+        <a href="#">{{ job.city | capitalize }}, {{ job.country }}</a> /
+        <a href="#">{{ job.contractType }}</a>
+
+        <br/><span class="archived">Archived since {{ job.createdAt | date('m/d/Y') }}</span>
```
Comment (mmarchois) [inline src/SensioLabs/JobBoardBundle/Resources/views/Includes/full_job_header.html.twig]: Remove spaces before and after filter

**Comment by Nicofuma** [comment]
You have some changes in common with your first PR. You should keep these changes in a separate commit to ease your life when rebasing (it also makes code review easier by isolating the changes per PRs)

File: src/SensioLabs/JobBoardBundle/Controller/JobController.php [hunk 7]
```diff
@@ -19,12 +20,26 @@ public function showAction()
...
+                        ->getRepository('SensioLabsJobBoardBundle:Job');
+
+        $job = $jobRepo->findOneBy(array(
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Controller/JobController.php]: if php 5.4 is supported, use short array syntaxe

File: src/SensioLabs/JobBoardBundle/Resources/public/css/custom.css [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+#error {
+    margin-bottom: 20px;
+}
+    #error > ul {
+        margin-bottom: 0;
+    }
+
+        #error > ul > li {
+            color: #F00;
+        }
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Resources/public/css/custom.css]: Missing end line

File: src/SensioLabs/JobBoardBundle/Resources/views/Includes/job.html.twig [hunk 9]
```diff
@@ -8,17 +8,19 @@
...
         <div class="clear"></div>
         <div id="job-description" class="padding description">
-            Lorem ipsum dolor sit amet, consectetur adipiscing elit. In quis nulla augue. Phasellus enim eros, luctus a dapibus ut, auctor et urna. Donec eget egestas mi. Sed auctor semper elit, quis elementum orci pulvinar ut. Donec leo quam, elementum non vulputate ut, laoreet eu augue. Pellentesque tempus luctus porttitor. In id orci enim, quis gravida massa. Praesent consequat hendrerit consequat. Cras nulla elit, pulvinar a dapibus vitae, pellentesque vel orci. Fusce tellus turpis, tempor eu convallis sed, scelerisque bibendum turpis. Fusce pharetra venenatis augue at condimentum. Duis lectus arcu, aliquam ut euismod non, gravida id nisl. Nullam hendrerit condimentum lacus in egestas. Morbi lobortis nibh ac lacus convallis adipiscing. Integer lacinia justo et nunc faucibus in pellentesque dui sagittis. Mauris mi felis, consequat ac laoreet pretium, tempor quis enim. Fusce arcu massa, posuere ut luctus ut, pellentesque a augue. 
+            {{ job.description }}
         </div>
     {% endblock %}
 
+    {% if job.howToApply %}
     {% block job_apply %}
```
Comment (jderusse) [inline src/SensioLabs/JobBoardBundle/Resources/views/Includes/job.html.twig]: ident nesteed blocks

File: src/SensioLabs/JobBoardBundle/Entity/Job.php [hunk 10]
```diff
@@ -75,6 +75,22 @@ class Job
     private $contractType;
 
     /**
+     * @var \Datetime $createdAt
+     *
+     * @Gedmo\Timestampable(on="create")
+     * @ORM\Column(name="created_at", type="datetime", nullable=true)
```
Comment (Nicofuma) [inline src/SensioLabs/JobBoardBundle/Entity/Job.php]: Why can it is nullable?

File: src/SensioLabs/JobBoardBundle/Entity/Job.php [hunk 11]
```diff
@@ -90,6 +106,12 @@ class Job
     private $howToApply;
 
     /**
+     * @Gedmo\Slug(fields={"company", "title", "id"}, updatable=false)
```
Comment (Nicofuma) [inline src/SensioLabs/JobBoardBundle/Entity/Job.php]: AS far as I know only the title should be included in the slug

**Pull request merged**